### PR TITLE
THEBIGBUGFIX: some whitespace fixes

### PR DESCRIPTION
**Pull request by taylortom**

### CODE REVIEW

File: frontend/src/core/editor/page/models/editorBlockModel.js [hunk 1]
```diff
@@ -1,29 +1,38 @@
...
-	});
-
-	return EditorBlockModel;
-
+  var EditorModel = require('editorGlobal/models/editorModel');
+
+  var EditorBlockModel = EditorModel.extend({
+    urlRoot: '/api/content/block',
+    _parent: 'articles',
+  	_siblings:'blocks',
```
Comment (canstudios-nicolaw) [inline frontend/src/core/editor/page/models/editorBlockModel.js]: Can this be moved in a level of indentation ? (Sorry to be picky)

File: frontend/src/core/app/router.js [hunk 2]
```diff
@@ -28,11 +26,10 @@ define(function(require) {
     },
 
     redirectToLogin: function() {
-       this.navigate('#/user/login', {trigger: true});
+      this.navigate('#/user/login', {trigger: true});
```
Comment (tomgreenfield) [inline frontend/src/core/app/router.js]: `{trigger: true}` → `{ trigger: true }`

File: frontend/src/core/dashboard/views/dashboardSidebarView.js [hunk 3]
```diff
@@ -1,179 +1,157 @@
...
+      }
+      this.highlightSearchBox();
+      if (userPreferences.tags) {
+        this.tags = userPreferences.tags;
+        _.each(userPreferences.tags, this.addTagToSidebar, this);
+      }
+    },
+
+    addCourse: function() {
+      Origin.router.navigate('#/project/new', {trigger:true});
```
Comment (tomgreenfield) [inline frontend/src/core/dashboard/views/dashboardSidebarView.js]: `{trigger:true}` → `{ trigger: true }`

File: frontend/src/core/dashboard/views/dashboardSidebarView.js [hunk 4]
```diff
@@ -1,179 +1,157 @@
...
+    },
+
+    gotoMyCourses: function() {
+      Origin.router.navigate('#/dashboard', {trigger: true});
```
Comment (tomgreenfield) [inline frontend/src/core/dashboard/views/dashboardSidebarView.js]: `{trigger: true}` → `{ trigger: true }`

File: frontend/src/core/dashboard/views/dashboardSidebarView.js [hunk 5]
```diff
@@ -1,179 +1,157 @@
...
+    },
+
+    gotoSharedCourses: function() {
+      Origin.router.navigate('#/dashboard/shared', {trigger: true});
```
Comment (tomgreenfield) [inline frontend/src/core/dashboard/views/dashboardSidebarView.js]: `{trigger: true}` → `{ trigger: true }`

File: frontend/src/core/editor/global/models/editorModel.js [hunk 6]
```diff
@@ -1,141 +1,118 @@
...
+    loadedData: function() {
+      if (this._siblings) {
+        this._type = this._siblings;
+      }
+      Origin.trigger('editorModel:dataLoaded', this._type, this.get('_id'));
+    },
+
+    getChildren: function() {
+      if (Origin.editor.data[this._children]) {
+        var children = Origin.editor.data[this._children].where({_parentId:this.get("_id")});
```
Comment (tomgreenfield) [inline frontend/src/core/editor/global/models/editorModel.js]: `{_parentId:this.get("_id")}` → `{ _parentId: this.get('_id') }`

File: frontend/src/core/editor/global/models/editorModel.js [hunk 7]
```diff
@@ -1,141 +1,118 @@
...
-            var siblings = Origin.editor.data[this._siblings].where({_parentId:this.get("_parentId")});
-            var siblingsCollection = new Backbone.Collection(siblings);
+      if (currentType === 'menu' || currentType === 'page') {
+        if (currentParentId === Origin.editor.data.course.get('_id')) {
+          parent = Origin.editor.data.course;
+        } else {
+          parent = Origin.editor.data.contentObjects.findWhere({_id: currentParentId});
+        }
+      } else if (currentType != 'course'){
+        parent = Origin.editor.data[this._parent].findWhere({_id: currentParentId});
```
Comment (tomgreenfield) [inline frontend/src/core/editor/global/models/editorModel.js]: Lines 42 & 45:
`{_id: currentParentId}` → `{ _id: currentParentId }`

File: frontend/src/core/editor/global/models/editorModel.js [hunk 8]
```diff
@@ -1,141 +1,118 @@
...
-            }, this));
-            var siblingsCollection = new Backbone.Collection(siblings);
+    },
 
-            
-            // returns a collection of siblings
-            return siblingsCollection;
+    getSiblings: function(returnMyself) {
+      if (returnMyself) {
+        var siblings = Origin.editor.data[this._siblings].where({ _parentId: this.get("_parentId") });
```
Comment (tomgreenfield) [inline frontend/src/core/editor/global/models/editorModel.js]: `"` → `'`

File: frontend/src/core/editor/global/models/editorModel.js [hunk 9]
```diff
@@ -1,141 +1,118 @@
...
+        return new Backbone.Collection(siblings);
+      }
+      var siblings = _.reject(Origin.editor.data[this._siblings].where({
+        _parentId:this.get("_parentId")
```
Comment (tomgreenfield) [inline frontend/src/core/editor/global/models/editorModel.js]: `_parentId:this.get("_parentId")` → `_parentId: this.get('_parentId')`

File: lib/helpers.js [hunk 10]
```diff
@@ -143,46 +138,43 @@ function isMasterPreviewAccessible(courseId, userId, next) {
...
+      : contentItem._id.toString();
+
+  database.getDatabase(function(err, db) {
+    if (err) {
+      logger.log('error', err);
+      return err;
+    }
 
-    database.getDatabase(function(err, db) {
+    db.retrieve('course', { _id: courseId }, {jsonOnly: true}, function (err, results) {
```
Comment (tomgreenfield) [inline lib/helpers.js]: `{jsonOnly: true}` → `{ jsonOnly: true }`

**Comment by taylortom** [comment]
@tomgreenfield Fixed. WHO'S NEXT? 👊

**Comment by hmaffin** [comment]
You're on fire Tom! :fire: :fire_engine:

File: frontend/src/core/dashboard/views/dashboardSidebarView.js [hunk 11]
```diff
@@ -1,179 +1,157 @@
...
+      }
+      this.highlightSearchBox();
+      if (userPreferences.tags) {
+        this.tags = userPreferences.tags;
+        _.each(userPreferences.tags, this.addTagToSidebar, this);
+      }
+    },
+
+    addCourse: function() {
+      Origin.router.navigate('#/project/new', { trigger:true });
```
Comment (tomgreenfield) [inline frontend/src/core/dashboard/views/dashboardSidebarView.js]: `trigger: true`

File: frontend/src/core/editor/page/models/editorArticleModel.js [hunk 12]
```diff
@@ -1,17 +1,17 @@
 // LICENCE https://github.com/adaptlearning/adapt_authoring/blob/master/LICENSE
 define(function(require) {
+  var EditorModel = require('editorGlobal/models/editorModel');
 
-	var EditorModel = require('editorGlobal/models/editorModel');
+  var EditorArticleModel = EditorModel.extend({
+    urlRoot: '/api/content/article',
+    _parent:'contentObjects',
+    _siblings:'articles',
```
Comment (tomgreenfield) [inline frontend/src/core/editor/page/models/editorArticleModel.js]:
```js
_parent: 'contentObjects',
_siblings: 'articles',
```

File: frontend/src/core/editor/page/models/editorBlockModel.js [hunk 13]
```diff
@@ -1,29 +1,38 @@
...
-	});
-
-	return EditorBlockModel;
-
+  var EditorModel = require('editorGlobal/models/editorModel');
+
+  var EditorBlockModel = EditorModel.extend({
+    urlRoot: '/api/content/block',
+    _parent: 'articles',
+    _siblings:'blocks',
```
Comment (tomgreenfield) [inline frontend/src/core/editor/page/models/editorBlockModel.js]: `_siblings: 'blocks'`

File: frontend/src/core/editor/page/models/editorComponentModel.js [hunk 14]
```diff
@@ -1,21 +1,38 @@
 // LICENCE https://github.com/adaptlearning/adapt_authoring/blob/master/LICENSE
 define(function(require) {
+  var EditorModel = require('editorGlobal/models/editorModel');
 
-    var EditorModel = require('editorGlobal/models/editorModel');
+  var EditorComponentModel = EditorModel.extend({
+    urlRoot: '/api/content/component',
+    _parent: 'blocks',
+    _siblings:'components',
```
Comment (tomgreenfield) [inline frontend/src/core/editor/page/models/editorComponentModel.js]: `_siblings: 'components'`

File: frontend/src/core/project/views/sharedProjectView.js [hunk 15]
```diff
@@ -30,28 +25,25 @@ define(function(require){
...
+    selectProject: function(e) {
+      e && e.preventDefault();
       this.selectItem();
     },
 
     selectItem: function() {
       Origin.trigger('dashboard:projectView:itemSelected');
       this.$el.addClass('selected');
-      this.model.set({_isSelected:true});
+      this.model.set({ _isSelected:true });
```
Comment (tomgreenfield) [inline frontend/src/core/project/views/sharedProjectView.js]: `_isSelected: true`

File: lib/auth.js [hunk 16]
```diff
@@ -179,7 +180,7 @@ exports = module.exports = {
    * @return {object} preloader - an AdaptBuilder ModulePreloader
    */
   preload: function (app) {
-    var preloader = new app.ModulePreloader(app, MODNAME, {events:preloadHandle(app,this)});
+    var preloader = new app.ModulePreloader(app, MODNAME, { events:preloadHandle(app,this) });
```
Comment (tomgreenfield) [inline lib/auth.js]: `events: preloadHandle(app, this)`

File: frontend/src/core/editor/global/models/editorModel.js [hunk 17]
```diff
@@ -1,141 +1,116 @@
...
+    loadedData: function() {
+      if (this._siblings) {
+        this._type = this._siblings;
+      }
+      Origin.trigger('editorModel:dataLoaded', this._type, this.get('_id'));
+    },
+
+    getChildren: function() {
+      if (Origin.editor.data[this._children]) {
+        var children = Origin.editor.data[this._children].where({ _parentId:this.get('_id') });
```
Comment (tomgreenfield) [inline frontend/src/core/editor/global/models/editorModel.js]: `_parentId: this.get('_id')`

File: frontend/src/core/project/views/sharedProjectView.js [hunk 18]
```diff
@@ -30,28 +25,25 @@ define(function(require){
...
     },
 
     deselectItem: function() {
       this.$el.removeClass('selected');
-      this.model.set({_isSelected:false});
+      this.model.set({ _isSelected:false });
```
Comment (tomgreenfield) [inline frontend/src/core/project/views/sharedProjectView.js]: `_isSelected: false`

**Pull request merged**